src/distributions/dragonwell/installer.ts

```diff
@@ -0,0 +1,173 @@
+        : '';
+      throw new Error(
+        `Cannot find satisfied version for ${version}.${versionsMsg}`
+      );
+    }
+    return resolvedVersion;
+  }
+
+  private async getAvailableVersions(): Promise<IDragonwellVersions[]> {
+    const platform = this.getPlatformOption();
+    let arch = this.distributionArchitecture();
+
+    const majorVersion = this.version.includes('.')
+      ? this.version.split('.')[0]
+      : this.version;
+    if (['8', '11', '17'].includes(majorVersion) != true) {
+      throw new Error('Support dragonwell versions: 8, 11, 17');
+    }
+
+    arch = arch.includes('x86') ? 'x64' : arch;
+
+    const availableVersionsUrl =
+      'https://dragonwell-jdk.io/map_with_checksum.json';
+
+    const fetchDragonwellVersions =
+      (await this.http.getJson<IDragonwellAllVersions>(availableVersionsUrl))
+        .result ?? {};
+    if (!fetchDragonwellVersions) {
+      throw Error(
+        `Could fetch any dragonwell versions from ${availableVersionsUrl}`
+      );
+    }
+    const availableVersions = this.getEligibleAvailableVersions(
+      platform,
+      arch,
+      fetchDragonwellVersions
+    );
+
+    if (core.isDebug()) {
+      core.startGroup('Print information about available versions');
+      core.endGroup();
+    }
+
+    return availableVersions;
+  }
+
+  protected async downloadTool(
+    javaRelease: JavaDownloadRelease
+  ): Promise<JavaInstallerResults> {
+    core.info(
+      `Downloading Java ${javaRelease.version} (${this.distribution}) from ${javaRelease.url} ...`
+    );
+    const javaArchivePath = await tc.downloadTool(javaRelease.url);
+
+    core.info(`Extracting Java archive...`);
+
+    const extractedJavaPath = await extractJdkFile(
+      javaArchivePath,
+      getDownloadArchiveExtension()
+    );
+
+    const archiveName = fs.readdirSync(extractedJavaPath)[0];
+    const archivePath = path.join(extractedJavaPath, archiveName);
+    const version = this.getToolcacheVersionName(javaRelease.version);
+
+    const javaPath = await tc.cacheDir(
+      archivePath,
+      this.toolcacheFolderName,
+      version,
+      this.architecture
+    );
+
+    return {version: javaRelease.version, path: javaPath};
+  }
+
+  private getEligibleAvailableVersions(
+    platform: string,
+    arch: string,
+    dragonwellVersions: IDragonwellAllVersions
+  ): IDragonwellVersions[] {
+    const eligibleVersions: IDragonwellVersions[] = [];
+
+    for (const majorVersion in dragonwellVersions) {
+      const majorVersionMap = dragonwellVersions[majorVersion];
+      for (let jdkVersion in majorVersionMap) {
+        const jdkVersionMap = majorVersionMap[jdkVersion];
+        if (!(platform in jdkVersionMap)) {
+          continue;
+        }
+        const platformMap = jdkVersionMap[platform];
+        if (!(arch in platformMap)) {
+          continue;
+        }
+        const archMap = platformMap[arch];
+        if (jdkVersion == 'latest') {
```

Comment:
```suggestion
        if (jdkVersion === 'latest') {
```
I'd suggest strict equality sign here.